src/test/java/com/microsoft/sqlserver/jdbc/bvt/bvtTest.java

```diff
@@ -1,678 +1,600 @@
+            rs2.verifyCurrentRow(table2);
+            rs1.next();
+            rs1.verifyCurrentRow(table1);
+            rs1.verify(table1);
+            rs1.close();
+            rs2.next();
+            rs2.verify(table2);
+        }
+        finally {
+            if (null != rs1) {
+                rs1.close();
+            }
+            if (null != rs2) {
+                rs2.close();
+            }
+            if (null != stmt1) {
+                stmt1.close();
+            }
+            if (null != stmt2) {
+                stmt2.close();
+            }
+            terminateVariation();
+        }
+    }
+
+    /**
+     * Verify two concurrent resultsets from same connection, same statement
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testTwoResultsetsSameStmt() throws SQLException {
+
+        DBResultSet rs1 = null;
+        DBResultSet rs2 = null;
+        try {
+            conn = new DBConnection(connectionString);
+            stmt = conn.createStatement();
+
+            String query = "SELECT * FROM " + table1.getEscapedTableName();
+            rs1 = stmt.executeQuery(query);
+
+            String query2 = "SELECT * FROM " + table2.getEscapedTableName();
+            rs2 = stmt.executeQuery(query2);
+
+            // Interleave resultset calls. rs is expected to be closed
+            try {
+                rs1.next();
+            }
+            catch (SQLException e) {
+                assertEquals(e.toString(), "com.microsoft.sqlserver.jdbc.SQLServerException: The result set is closed.");
+            }
+            rs2.next();
+            rs2.verifyCurrentRow(table2);
+            try {
+                rs1.next();
+            }
+            catch (SQLException e) {
+                assertEquals(e.toString(), "com.microsoft.sqlserver.jdbc.SQLServerException: The result set is closed.");
+            }
+            rs1.close();
+            rs2.next();
+            rs2.verify(table2);
+        }
+        finally {
+            if (null != rs1) {
+                rs1.close();
+            }
+            if (null != rs2) {
+                rs2.close();
+            }
+            terminateVariation();
+        }
+    }
+
+    /**
+     * Verify resultset closed after statement is closed
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testResultSetAndCloseStmt() throws SQLException {
+        try {
+            conn = new DBConnection(connectionString);
+            stmt = conn.createStatement();
+
+            String query = "SELECT * FROM " + table1.getEscapedTableName();
+            rs = stmt.executeQuery(query);
+
+            stmt.close(); // this should close the resultSet
+            try {
+                rs.next();
+            }
+            catch (SQLException e) {
+                assertEquals(e.toString(), "com.microsoft.sqlserver.jdbc.SQLServerException: The result set is closed.");
+            }
+        }
+        finally {
```

Comment:
it will be good if we can also have an assert after catch, to verify code execution dint reach that part of code.